From 90a1430fca8d5165e1909de9f009aa9f4d6430ef Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Wed, 29 Nov 2017 18:36:17 +0100
Subject: [PATCH] link_gre6: Detect invalid encaplimit values

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1459600
Upstream Status: iproute2.git commit 56708ae7c9535

commit 56708ae7c9535859223c5b68097b35bf0fae677c
Author: Phil Sutter <phil@nwl.cc>
Date:   Tue Nov 28 16:49:58 2017 +0100

    link_gre6: Detect invalid encaplimit values

    Looks like a typo: get_u8() returns 0 on success and -1 on error, so the
    error checking here was ineffective.

    Fixes: a11b7b71a6eba ("link_gre6: really support encaplimit option")
    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 ip/link_gre6.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ip/link_gre6.c b/ip/link_gre6.c
index 932f9ee96124d..a9d18ee954641 100644
--- a/ip/link_gre6.c
+++ b/ip/link_gre6.c
@@ -351,7 +351,7 @@ get_failed:
 			} else {
 				__u8 uval;
 
-				if (get_u8(&uval, *argv, 0) < -1)
+				if (get_u8(&uval, *argv, 0))
 					invarg("invalid ELIM", *argv);
 				encap_limit = uval;
 				flags &= ~IP6_TNL_F_IGN_ENCAP_LIMIT;
-- 
2.21.0

